src/routes.rs

```diff
@@ -50,6 +57,15 @@ pub async fn get_vehicle_data(State(app): State<Arc<AppState>>) -> Json<VehicleR
     })
 }
 
+pub async fn get_grue_data(Path(id): Path<u8>, State(app): State<Arc<AppState>>) -> impl IntoResponse {
+    match app.get_specific_grue_date(id).await {
```

Comment:
Here, you will need to pass a verified `LoadingZone` instead on a primitive `u8`.